core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -199,21 +198,29 @@ private void initializeCatalogTables() throws InterruptedException, SQLException
         });
   }
 
-  private void updateCatalogTables(Connection connection) throws SQLException {
-    DatabaseMetaData dbMeta = connection.getMetaData();
-    ResultSet typeColumn =
-        dbMeta.getColumns(null, null, JdbcUtil.CATALOG_TABLE_VIEW_NAME, JdbcUtil.RECORD_TYPE);
-    if (typeColumn.next()) {
-      LOG.debug("{} already supports views", JdbcUtil.CATALOG_TABLE_VIEW_NAME);
-    } else {
-      if (PropertyUtil.propertyAsBoolean(
-          catalogProperties, JdbcUtil.ADD_VIEW_SUPPORT_PROPERTY, false)) {
-        connection.prepareStatement(JdbcUtil.UPDATE_CATALOG_SQL).execute();
-      } else {
-        LOG.warn(VIEW_WARNING_LOG_MESSAGE);
-        schemaVersion = JdbcUtil.SchemaVersion.V0;
-      }
-    }
+  private void updateCatalogTables() throws SQLException, InterruptedException {
```

Comment:
what about naming this `checkAndMaybeUpdateSchema` (or anything else). `updateCatalogTables` doesn't fully represent what this method is doing internally

Reply:
Good point, let me rename and add a couple of tests.

Reply:
also we should have a test where the catalog is initialized with `initializeCatalogTables = false`. That test would then initialize legacy schema and read and create tables (similar to `testLegacySchemaSupport()`)

Reply:
Agree, I'm working on it now.

Reply:
I added a `ParameterizedTest` test to test with `initializeCatalogTables` `true|false`.